Workflow/Sources/WorkflowNode.swift

```diff
@@ -48,9 +48,19 @@ final class WorkflowNode<WorkflowType: Workflow> {
         let output: Output
 
         switch subtreeOutput {
-        case .update(let event, let source):
-            /// Apply the update to the current state
-            let outputEvent = event.apply(toState: &state)
+        case .update(let action, let source):
+
+            /// 'Opens' the existential `any WorkflowAction<WorkflowType>` value
+            /// allowing the underlying conformance to be applied to the Workflow's State
+            func openAndApply<A: WorkflowAction>(
```

Comment:
it's not entirely clear to me why this is necessary, or exactly why it works, but if you try to invoke `action.apply(...)` on the `any WorkflowAction` variable, you get a compiler error. this makes some amount of sense b/c an `any WorkflowAction` isn't itself a `WorkflowAction`. i did skim [this](https://github.com/apple/swift-evolution/blob/main/proposals/0352-implicit-open-existentials.md) and [this](https://forums.swift.org/t/using-openexistential-to-construct-a-rawrepresentable-type/48288/4) to see if there are other options, but this was the only thing i've been able to get working thus far.